Unable to build current master (bisected)

**Describe the bug**
Current master doesn't build.

```
[ 33%] Building C object bin/CMakeFiles/lib_darktable.dir/common/imagebuf.c.o
/home/jade/.local/git/darktable/src/common/exif.cc: In function 'int dt_exif_xmp_attach_export(int, const char*, void*)':
/home/jade/.local/git/darktable/src/common/exif.cc:4009:29: error: 'kerTooLargeJpegSegment' is not a member of 'Exiv2'
       if(e.code() == Exiv2::kerTooLargeJpegSegment)
                             ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
make[2]: *** [bin/CMakeFiles/lib_darktable.dir/build.make:368: bin/CMakeFiles/lib_darktable.dir/common/exif.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:1592: bin/CMakeFiles/lib_darktable.dir/all] Error 2
make: *** [Makefile:152: all] Error 2
```

**To Reproduce**
1. Pull/clone
2. Start build process
3. Doesn't finish

**Expected behavior**
To build cleanly.

**Bisect**

```
[stasis] jade ~/.local/git/darktable $ git bisect good
6c3ea20c3961fd924bd49dc973e0b3dd9a80a515 is the first bad commit
commit 6c3ea20c3961fd924bd49dc973e0b3dd9a80a515
Author: phweyland <philippe.weyland@laposte.net>
Date:   Sat Jan 23 12:44:22 2021 -0300

    export metadata: remove history from xmp if max size exceeded
    
    fixes #4355

:040000 040000 bc1a009385273bb844f2f035aaa1fca2a554a981 f05a7ec8b15a89cfa3e8e1d3fcf05aa0b0930892 M      src
```

* darktable version : 3.5.0+817 gba3ce374e / 3.4.0 

**Platform**

* OS                : Linux (kernel 5.8.10-1)
* Distro            : Debian Buster (rel. 10.7)
* Graphics driver   : nvidia (450.80.02)
* OpenCL installed  : Yes (nvidia-opencl-common 450.80.02-1)
* OpenCL activated  : Yes
* Xorg              : 7.7+19

**Build info**

* Compiler      : gcc 8.3.0
* Cmake         : 3.13.4
* Gtk+          : 3.24.5-1
* Glib          : 2.58.3-2
* Sqlite        : 3.27.2
* Optimization  : None

**Hardware**

* CPU              : AMD Ryzen 5 3600 6-Core Processor
* GPU              : GeForce RTX 2060 SUPER 7979 MiB
* Memory           : 32 Gb + 15 Gb Swap

**Additional context**

[dt.build.2021.01.25.0859.log](https://github.com/darktable-org/darktable/files/5868145/dt.build.2021.01.25.0859.log)

Ping @phweyland 

EDIT: Added correct logfile

--- a/cmake/modules/FindExiv2.cmake
+++ b/cmake/modules/FindExiv2.cmake
@@ -44,6 +44,10 @@ if(Exiv2_PKGCONF_VERSION VERSION_LESS Exiv2_FIND_VERSION)
   message(FATAL_ERROR "Exiv2 version check failed.  Version ${Exiv2_PKGCONF_VERSION} was found, at least version ${Exiv2_FIND_VERSION} is required")
 endif(Exiv2_PKGCONF_VERSION VERSION_LESS Exiv2_FIND_VERSION)
 
+if (Exiv2_PKGCONF_VERSION)
+  set(Exiv2_VERSION "${Exiv2_PKGCONF_VERSION}")
+endif()
+
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(Exiv2 DEFAULT_MSG Exiv2_LIBRARY Exiv2_INCLUDE_DIR)
 

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -379,6 +379,12 @@ endforeach(lib)
 
 # Require exiv2 >= 0.24 to make sure everything we need is available
 find_package(Exiv2 0.24 REQUIRED)
+if(NOT Exiv2_VERSION VERSION_LESS 0.27)
+  add_definitions("-DHAVE_EXIV2_027_OR_NEWER")
+  message(STATUS "Exiv2 version 0.27 or newer")
+else()
+  message(STATUS "Exiv2 version older than 0.27, some features may be downgraded")
+endif()
 include_directories(SYSTEM ${Exiv2_INCLUDE_DIRS})
 list(APPEND LIBS ${Exiv2_LIBRARIES})
 add_definitions(${Exiv2_DEFINITIONS})

--- a/src/common/exif.cc
+++ b/src/common/exif.cc
@@ -4006,7 +4006,11 @@ int dt_exif_xmp_attach_export(const int imgid, const char *filename, void *metad
     }
     catch(Exiv2::AnyError &e)
     {
+#ifdef HAVE_EXIV2_027_OR_NEWER
       if(e.code() == Exiv2::kerTooLargeJpegSegment)
+#else
+      if(e.code() == 37)
+#endif
       {
         _remove_xmp_keys(xmpData, "Xmp.darktable.history");
         _remove_xmp_keys(xmpData, "Xmp.darktable.masks_history");

